nucypher/blockchain/eth/actors.py

```diff
@@ -585,6 +594,83 @@ def encrypt_threshold_decryption_response(
             requester_public_key=requester_public_key,
         )
 
+
+    def _authorize_ritual(self, decryption_request: ThresholdDecryptionRequest):
+        # TODO: #3052 consider using the DKGStorage cache instead of the coordinator agent
+        # dkg_public_key = this_node.dkg_storage.get_public_key(decryption_request.ritual_id)
+        ritual = self.coordinator_agent.get_ritual(
+            decryption_request.ritual_id, with_participants=True
+        )
+
+        # enforces that the node is part of the ritual
+        participants = [p.provider for p in ritual.participants]
+        if self.checksum_address not in participants:
+            raise self.UnauthorizedRequest(
+                f"Node not part of ritual {decryption_request.ritual_id}",
+            )
+
+        # check that ritual not timed out
+        current_timestamp = self.coordinator_agent.blockchain.get_blocktime()
+        if current_timestamp > ritual.end_timestamp:
+            raise self.UnauthorizedRequest(
+                f"Ritual {decryption_request.ritual_id} is expired",
+            )
+
+    def _authorize_ciphertext(
```

Comment:
This sounds like an action vs a check. Perhaps something like `_check_ciphertext_authorization(...)`?

Reply:
Naming things... 😅 I can see how it might sound confusing since we use "authorize" to mean granting access via the public API. In a sense this the moment when enrico's authorization is actually enforced: ursula is guarding it's private materials and grants the request to use the cryptographic backend.  To me that sounds like an "authorization"... but perhaps "verify"?  it invokes cryptographic ideas.

```python
def _verify_ciphertext_authorization(...)
```

Reply:
👌

Reply:
✔️